api/app/controllers/api/v1/comments_controller.rb

```diff
@@ -0,0 +1,35 @@
+module Api
+  module V1
+    class CommentsController < BaseController
+      before_action :validate_params!
+
+      def create
+        comment = current_user.comments.build(content: comment_params[:content], parent:)
+        return render json: ::V1::CommentsBlueprint.render(comment) if comment.save
+
+        render json: { errors: comment.errors.full_messages }, status: :unprocessable_entity
+      end
+
+      private
+
+      def validate_params!
+        return head :bad_request if comment_params[:content].blank?
+        return head :bad_request if comment_params[:parent_uuid].blank?
+        return head :bad_request if comment_params[:parent_type].blank?
+        return head :bad_request unless %w[Issue Comment].include?(comment_params[:parent_type])
+      end
+
+      def comment_params
+        params.permit(:content, :parent_uuid, :parent_type)
+      end
+
+      def parent
+        @parent ||= if comment_params[:parent_type] == 'Issue'
+                      Issue.find_by(uuid: comment_params[:parent_uuid])
+                    else
+                      Comment.find_by(uuid: comment_params[:parent_uuid])
+                    end
+      end
```

Comment:
```suggestion
      def parent
        @parent ||= issue_or_comment
      end
      
      def issue_or_comment
        parent_type = comment_params[:parent_type]
        return Issue.find_by(uuid: comment_params[:parent_uuid]) if parent_type == 'Issue'
        
        Comment.find_by(uuid: comment_params[:parent_uuid])
      end
```

Reply:
what do you think

Reply:
good call. I digged a bit more into it and decided to move on with this option
```ruby
      def parent
        @parent ||= comment_params[:parent_type].constantize.find_by(uuid: comment_params[:parent_uuid])
      end
```